Makefile

```diff
@@ -0,0 +1,36 @@
+GO ?= go
+GH_ORG ?= Azure-Samples
+ROOT = github.com/$(GH_ORG)/azure-sdk-for-go-samples
+BASE = $(GOPATH)/src/$(ROOT)
+
+# tested on 1.9+, no need to exclude /vendor
+# to specify packages to skip: `PKGS_SKIP_RE='storage|graphrbac' make ...`
+PKGS_SKIP_RE ?= ''
+PKGS         != $(GO) list ./... | sed 's@$(ROOT)/@./@' | grep -v -E "$(PKGS_SKIP_RE)"
+
+# uses Azure resources
+test: dep
+	$(GO) test -v $(PKGS)
+
+test_pr: lint
+	$(GO) build -v \
+		$(shell $(GO) list ./... | sed 's@$(ROOT)/@./@' | grep -v -E "$(PKGS_SKIP_RE)")
+
+dep:
+	$(GO) get -u github.com/golang/dep/cmd/dep
+	cd $(BASE) && dep ensure
+
+lint: dep
```

Comment:
I believe `gofmt` requires that the packages build. Instead of having `lint` depend on `dep`, I'd have it depend on a `build` phase.

Reply:
Okay, I'll split out a build phase. If all seems good I'll merge cause I have some stuff coming next :).

Reply:
Actually, I'm not sure it makes sense to build before lint, build may not work but you still want lint. If you don't mind I'll continue to have lint not depend on build for now, but we can experiment and change it.

Reply:
No worries!